### ISSUE TITLE
`-->` ignored in series recipes

### ISSUE THREAD

**Issue by mkborregaard** [JuliaPlots/Plots.jl#1639]
I noticed this as I was testing through all of StatPlots (it makes the ea_histogram recipe fail), but I also get it now even on very old (~1 year) versions of RecipesBase and Plots, and running on 0.6.3. If all users experience this, it is a big problem for the recipe ecosystem, so it is a priority. It is not 0.7-related.

```julia
@recipe function g(::Type{Val{:testhist}}, x, y, z)
        markercolor := :red
        seriestype := :scatter
        ()
end

@recipe function f(::Type{Val{:testhist2}}, x, y, z)
        markercolor --> :red
         seriestype := :scatter
         ()
end

plot(rand(10,2), st = [:testhist :testhist2], layout = 2)
```

<img width="591" alt="skaermbillede 2018-08-12 kl 16 55 26" src="https://user-images.githubusercontent.com/8429802/44003343-891a8502-9e51-11e8-858c-b708cbecfd98.png">

**Comment by tkf** [comment]
Is #1569 I reported before (when I was using Julia 0.6) related to this? My example was also using `-->`.

**Comment by mkborregaard** [comment]
No, #1569 is not the same, as using `:=` rather than `-->` does nothing.

**Comment by mkborregaard** [comment]
I've looked deeply into this - it turns out that this is because defaults are added BEFORE series recipes (but after user/type recipes) are processed - so all the values are defined and `-->` is dropped. See https://github.com/JuliaPlots/Plots.jl/blob/master/src/plot.jl#L226-L234 
I've looked at the file history, and to me it looks like it has always been this way - so indeed this probably never worked. But it should! The question is whether this can be refactored.
